### PR TITLE
Make multiplayer room listing filter by current ruleset

### LINKED ISSUE #7516: Multiplayer lounge does not filter rooms by current mode
**Describe the bug:** 

Multiplayer lounge does not filter rooms by current mode, which makes searching for rooms difficult.

**Screenshots or videos showing encountered issue:** 

Example with Taiko selected:

image

**osu!lazer version:** 2020.112.0

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on https://github.com/ppy/osu/pull/7739
- [x] Depends on #7741
- Closes https://github.com/ppy/osu/issues/7516

### CODE REVIEW

File: osu.Game/Screens/Multi/Multiplayer.cs [hunk 1]
```diff
@@ -32,8 +32,6 @@ public class Multiplayer : OsuScreen, IOnlineComponent
     {
         public override bool CursorVisible => (screenStack.CurrentScreen as IMultiplayerSubScreen)?.CursorVisible ?? true;
 
-        public override bool DisallowExternalBeatmapRulesetChanges => true;
```
Comment (smoogipoo) [inline osu.Game/Screens/Multi/Multiplayer.cs]: Why was this removed? This will cause the following:

```
System.InvalidOperationException: Attempted to lease a bindable that is already in a leased state.
   at osu.Framework.Bindables.Bindable`1.BeginLease(Boolean revertValueOnReturn)
   at osu.Game.Screens.OsuScreenDependencies..ctor(Boolean requireLease, IReadOnlyDependencyContainer parent) in /Users/smoogipoo/Repos/osu/osu.Game/Screens/OsuScreenDependencies.cs:line 30
   at osu.Game.Screens.OsuScreen.createDependencies(IReadOnlyDependencyContainer dependencies) in /Users/smoogipoo/Repos/osu/osu.Game/Screens/OsuScreen.cs:line 118
   at osu.Game.Screens.OsuScreen.CreateLeasedDependencies(IReadOnlyDependencyContainer dependencies) in /Users/smoogipoo/Repos/osu/osu.Game/Screens/OsuScreen.cs:line 101
   at osu.Game.Screens.OsuScreenStack.screenPushed(IScreen prev, IScreen next) in /Users/smoogipoo/Repos/osu/osu.Game/Screens/OsuScreenStack.cs:line 43
   at osu.Framework.Screens.ScreenStack.Push(IScreen source, IScreen newScreen)
   at osu.Framework.Screens.ScreenExtensions.<>c__DisplayClass0_0.<Push>b__0(ScreenStack stack)
   at osu.Framework.Screens.ScreenExtensions.runOnRoot(IDrawable current, Action`1 onRoot, Action onFail)
   at osu.Framework.Screens.ScreenExtensions.runOnRoot(IDrawable current, Action`1 onRoot, Action onFail)
   at osu.Framework.Screens.ScreenExtensions.Push(IScreen screen, IScreen newScreen)
   at osu.Game.Screens.Multi.Multiplayer.Start(Func`1 player) in /Users/smoogipoo/Repos/osu/osu.Game/Screens/Multi/Multiplayer.cs:line 164
   at osu.Game.Screens.Multi.Match.MatchSubScreen.onStart() in /Users/smoogipoo/Repos/osu/osu.Game/Screens/Multi/Match/MatchSubScreen.cs:line 226
```

File: osu.Game/Screens/Multi/Lounge/Components/FilterControl.cs [hunk 2]
```diff
@@ -22,6 +23,9 @@ public class FilterControl : SearchableListFilterControl<PrimaryFilter, Secondar
         [Resolved(CanBeNull = true)]
         private Bindable<FilterCriteria> filter { get; set; }
 
+        [Resolved]
+        private IBindable<RulesetInfo> ruleset { get; set; }
```
Comment (smoogipoo) [inline osu.Game/Screens/Multi/Lounge/Components/FilterControl.cs]: This doesn't actually seem to work with the game's ruleset selector.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Multi/Multiplayer.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Multi/Multiplayer.cs]: Ah. This is unfortunate and untested..

Comment (peppy) [reply osu.Game/Screens/Multi/Multiplayer.cs]: Fixed via #7741.

**Review by smoogipoo — APPROVED** [review]
